src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -48,7 +48,7 @@ public RepoConfiguration(RepoLocation location, String branch) {
     public RepoConfiguration(RepoLocation location, String branch, List<Format> formats, List<String> ignoreGlobList,
             boolean isStandaloneConfigIgnored, List<CommitHash> ignoreCommitList) {
         this.location = location;
-        this.branch = branch;
+        this.branch = location.isEmpty() ? "" : branch;
```

Comment:
Why is this necessary?

Reply:
I believe this is a kind of defensive programming because a `RepoConfiguration` with empty location do not need to have a branch. If an author config contains a line that has an empty location and non-empty branch, the equal() method of `RepoConfiguration` will fail and the author will not be added to the existing `AuthorConfig` that with empty location. The alternative way is to change the `equal()` method. But I believe doing this way is better.